lib/src/Collection.cpp

```diff
@@ -247,11 +247,24 @@ pplx::task<shared_ptr<Document>> Collection::CreateDocumentAsync(
 	{
 		value json_response = response.extract_json().get();
 
-		if (response.status_code() == status_codes::Created)
-		{
-			return DocumentFromJson(json_response);
-		}
+		for (size_t i = 0; i < MAX_QUERY_RETRY_CNT; ++i) {
+			if (response.status_code() == status_codes::Created)
+			{
+				return DocumentFromJson(json_response);
+			}
+			else if (429 == response.status_code()) {

```

Comment:
plesae no magic numbers, wouldn't status_codes::TooManyRequests work?

Reply:
Sorry. It was developed with cpprestsdk version 2.8.0, but it was not defined in the http_constants.dat file of that version, so I used numbers. Status_codes :: TooManyRequests will work fine in the latest cpprestsdk